aiida/tools/importexport/archive/zip_path.py

```diff
@@ -114,10 +117,20 @@ def _all_at_set(self) -> Set[str]:
         """Iterate through all file and directory paths in the zip file.
 
         Note: this is necessary, since the zipfile does not strictly store all directories in the namelist.
```

Comment:
Just wondering: so what is the zipfile policy for storing stuff in the namelist?
Is the problem that empty directories aren't stored?

Reply:
basically when you store a file (or directory) it just stores the path to that file in the index, not any of the parent directories. Then when extracting it makes sure the directories are created.
So if you want to check that a directory exists, you can't just look in the index, you have to also check parent directories